### ISSUE TITLE
Feature requests: OpenAL Soft & Stalkers dying in anomalies

### ISSUE THREAD

**Issue by ghost** [avoitishin/xray-16#28]
I realize you're not back to doing things yet as you're bored with Stalker but if/when you come back, I was thinking it would be nice to see OpenAL32 replaced with OpenAL Soft.

It has many benefits, including:
support for mono, stereo, 4-channel, 5.1, 6.1, 7.1, and HRTF output.
EFX for environmental audio effects.

I realize it would be a big undertaking but I think people would be able to appreciate the effort if they could pick true surround sound in-game in Stalker, or pick HRTF if using headphones.

I also realize that OpenAL Soft can work without being integrated, but I think most people won't bother to install it, and to make things more streamlined people would be able to pick these option in the game's sound menu.

---

The second feature I would like to see is NPC's being killed by anomalies. Both myself and Alundaio have looked through the code, comparing it to the SOC code, to see why NPC's aren't being damaged, but so far have not come up with a solution.

I did notice the following in the scripts for npcs and wonder if it's disabling damage from anom's:
new_action:add_precondition(world_property(stalker_ids.property_anomaly,false))

Also Alundaio found an issue in CAI_Stalker::inside_anomaly in ai_stalker_fire.cpp that may relate to it.

If you can figure it out it would be an amazing addition to Stalker:COP.

**Comment by revolucas** [comment]
No, adding stalker_ids.property_anomaly is not the cause. These are just preconditions for the action schemes. It means they wont' execute unless property_anomaly is false. The value (true or false) for property_inside_anomaly and property_anomaly are dictated by the function I told you about, inside_anomaly.

I have a feeling it something to do with the state_manager or something that is not in the lua scripts anymore.
